app/src/main/java/ru/thstdio/aa2020/cache/Repository.kt

```diff
@@ -0,0 +1,163 @@
+package ru.thstdio.aa2020.cache
+
+import android.content.Context
+import com.jakewharton.retrofit2.converter.kotlinx.serialization.asConverterFactory
+import kotlinx.coroutines.async
+import kotlinx.coroutines.coroutineScope
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.json.Json
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import ru.thstdio.aa2020.api.interceptor.ApiHeaderInterceptor
+import ru.thstdio.aa2020.api.response.*
+import ru.thstdio.aa2020.api.service.TimeDbApi
+import ru.thstdio.aa2020.cache.entity.*
+import ru.thstdio.aa2020.cache.relation.toCinema
+import ru.thstdio.aa2020.cache.relation.toCinemaDetail
+import ru.thstdio.aa2020.data.*
+import java.util.concurrent.atomic.AtomicReference
+
+
+const val ITEMS_SIZE_IN_PAGE = 20
+
+@ExperimentalSerializationApi
+class Repository(applicationContext: Context) {
+    private val api: TimeDbApi = createApi()
+    private val db: AppDb = AppDb.create(applicationContext)
+
+    private val configurationAtomic: AtomicReference<ConfigurationResponse> =
+        AtomicReference<ConfigurationResponse>()
+    private val genresAtomic: AtomicReference<Map<Long, Genre>> =
+        AtomicReference<Map<Long, Genre>>()
+
+
+    suspend fun getMoviesFromPage(page: Int): CinemaListWithTotalPage =
+        coroutineScope {
+            val configurationAndGenres = async { getConfigurationAndGenres() }
+            val responseAsync = async { api.getNowPlaying(page) }
+            val (configuration, genres) = configurationAndGenres.await()
+            val response = responseAsync.await()
+            async { saveCinemaListToDb(response.results, page, configuration, genres) }
```

Comment:
async { should be inside saveCinemaListToDb
u can also add Async postfix to function name